pages/api/auth/[...nextauth].js

```diff
@@ -0,0 +1,50 @@
+import NextAuth from "next-auth";
+import Providers from "next-auth/providers";
+import Adapters from "next-auth/adapters";
+import { PrismaClient } from "@prisma/client";
+
+const prisma = new PrismaClient();
+
+export default NextAuth({
+    // site URL for authentication redirect
+    site: process.env.NEXTAUTH_URL,
+    providers: [
+        // email provider with magic links
+        Providers.Email({
+            // AWS SES mail server
+            server: process.env.EMAIL_SERVER,
+            // email address from which the email is to be sent
+            from: process.env.EMAIL_FROM,
+            // maximum life of email magic link - 1 hour
+            maxAge: 24 * 60,
+            // TODO: add custom email handler
+        }),
+    ],
+    // map database adapter to prisma client
+    adapter: Adapters.Prisma.Adapter({
+        prisma,
+        modelMapping: {
+            User: "user",
+            Account: "account",
+            Session: "session",
+            VerificationRequest: "verificationRequest",
+        },
+    }),
+    pages: {
+        // On errors, redirect to home
+        error: "/",
+        // On email verification request, redirect to verify page
+        verifyRequest: "/verify",
+    },
+    session: {
+        // use JSON web tokens for session handling
+        jwt: true,
+        // maximum age of an idle session - 30 days
+        maxAge: 30 * 24 * 60 * 60,
+    },
+    jwt: {
+        // JWT secret used for generating a key
+        secret: process.env.JWT_SECRET,
+    },
+    // TODO: define custom callback functions
```

Comment:
Do we also have to pass database config too? Will that be done at a later time?

Reply:
We actually don't need to do this because we have a configuration for the prisma database adapter. NextAuth will just grab the database config from the prisma adapter.